src/charm.py

```diff
@@ -90,13 +90,17 @@ def __init__(self, *args: typing.Any):
         """
         super().__init__(*args)
         self.haproxy_service = HAProxyService()
+        # Order is important here as we want _ensure_tls to check if the hostname is configured
+        self.framework.observe(self.on[TLS_CERT_RELATION].relation_created, self._ensure_tls)
```

Comment:
btw would make sense to add a handler to the relation broken?

Reply:
I don't think we need to do verification for `relation-broken` events ( the tls v4 library doesn't hook this event either ) Is there a specific scenario that you have in mind?

Reply:
I was thinking about wanting only http but now I remember that is required. So never mind ;)